drivers/can/can_esp32_twai_priv.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Newer ESP32-series MCUs like ESP32-C3 and ESP32-S2 have some slightly different registers
+ * compared to the original ESP32, which is fully compatible with the SJA1000 controller.
+ *
+ * This file adds register definitions for the incompatible registers so that the TWAI frontend
+ * can implement custom implementations (different from sja1000 backend) depending on the MCU.
+ */
+
+#ifndef ZEPHYR_DRIVERS_CAN_ESP32_TWAI_PRIV_H_
+#define ZEPHYR_DRIVERS_CAN_ESP32_TWAI_PRIV_H_
+
+#include "can_sja1000_priv.h"
+
+/*
+ * TWAI_BUS_TIMING_0_REG is incompatible to CAN_SJA1000_BTR0:
+ *   - TWAI_BAUD_PRESC has size of 13 bits instead of 6 bits
+ *   - TWAI_SYNC_JUMP_WIDTH is moved to BIT(14) instead of BIT(6)
+ */
```

Comment:
Move this comment to the custom `set_timing()` function to document why it is needed.